### ISSUE TITLE
v-slot="$slots" report vue/no-unused-vars

### ISSUE THREAD

**Issue by yoyo837** [vuejs/eslint-plugin-vue#1899]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:**  8.15.0
- **eslint-plugin-vue version:** 9
- **Node version:** 16.15.0
- **Operating System:** macOS Monterey

**Please show your full configuration:**

```json5
module.exports = {
  root: true,
  env: {
    node: true,
    es2021: true,
  },
  extends: [
    'eslint:recommended',
    'standard',
    // 'plugin:vue/recommended',
    'plugin:vue/vue3-recommended',
    'plugin:import/errors',
    'plugin:import/warnings',
    'plugin:compat/recommended',
    'plugin:unicorn/recommended',
    'plugin:prettier/recommended',
  ],
  plugins: ['module-resolver', '@babel'],
  parserOptions: {
    ecmaVersion: 2021,
    sourceType: 'module',
    parser: '@babel/eslint-parser',
    ecmaFeatures: {
      jsx: true,
    },
  },
  globals: {
    defineProps: true,
    defineEmits: true,
    defineExpose: true,
  },
  overrides: [
    {
      files: ['babel.config.js', '.eslintrc.js', 'vue.config.js', 'tailwind.config.js', 'script/**/*'],
      rules: {
        'unicorn/prefer-module': 0,
      },
    },
  ],
  settings: {
    'import/resolver': {
      'babel-module': {},
      webpack: {
        config: './node_modules/@vue/cli-service/webpack.config.js',
      },
    },
    polyfills: ['Promise'],
  },
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    eqeqeq: ['error', 'always', { null: 'ignore' }],
    // 强制要求js文件以外的文件import的时候有后缀
    'import/extensions': [
      'error',
      'always',
      {
        js: 'never',
        mjs: 'never',
      },
    ],
    '@babel/new-cap': 0,
    'unicorn/no-null': 0,
    'unicorn/prevent-abbreviations': 0,
    'unicorn/filename-case': 0,
    'unicorn/no-array-reduce': 0,
    'unicorn/no-array-for-each': 0,
    'unicorn/catch-error-name': 0,
    'import/no-duplicates': 2,
    '@babel/no-invalid-this': 0,
    '@babel/no-unused-expressions': 0,
    '@babel/object-curly-spacing': 0,
    '@babel/semi': 2,
    'module-resolver/use-alias': 2,
    'vue/require-explicit-emits': 2,
    'vue/require-default-prop': 0,
    'vue/multi-word-component-names': 0,
    'vue/attributes-order': 2,
    'vue/no-undef-components': [
      'error',
      {
        ignorePatterns: ['van(\\-\\w+)+', 'el(\\-\\w+)+'],
      },
    ],
  },
};

```

**What did you do?**	

```vue
<template>
  <van-loading
    v-slot="$slots"
    v-bind="$attrs"
  >
  </van-loading>
</template>
```

**What did you expect to happen?**

No lint error.

**What actually happened?**

It report  `$slots`: `vue/no-unused-vars`

**Repository to reproduce this issue**

**Comment by ota-meshi** [comment]
I think you have misunderstood how to use v-slot.
The rule is working correctly. Check the documentation.
https://vuejs.org/guide/components/slots.html#scoped-slots
